first try on adding Docker to help development

mikkeschiren opened the pull request:
Fixes #102

Commit:
first try on adding Docker to help development

joeyciechanowicz commented on Dockerfile:
```diff
@@ -0,0 +1,13 @@
+FROM node:10.19.0-stretch
```
You might want to use `[buildkite/puppeteer`](https://hub.docker.com/r/buildkite/puppeteer/tags/)` as I seem to recall it's a real pain getting the `node` image to run Puppeteer. (which Pa11y needs)

mikkeschiren replied:
True, puppeter is needed, but did a try on buildkite/puppeteer - but it did not work out with the sidekick code is it is right now - but I will look into it -I will be back on the subject.

mikkeschiren replied:
I updated Dockerfile with another setup that I am using with puppeteer, using alpine.

mikkeschiren replied:
Or rather - using puppeteer-core

joeyciechanowicz commented on docker-compose.yml:
```diff
@@ -0,0 +1,24 @@
+      LOG_LEVEL: debug
+      NODE_ENV: development
+      PORT: 8080
+      SESSION_SECRET: b3c542a1-79ce-46a1-b958-06e352eba249
```
Can this be read from the environment variables, as it should be different per installation. Think you can read in variables from the host shell using
```
      SESSION_SECRET: ${SESSION_SECRET}
```

mikkeschiren replied:
Yeah I guess that would be the best.

joeyciechanowicz reviewed (commented):
Great idea adding this, thankyou! Just a couple of comments

Commit:
now with puppeter core

Commit:
env variables

Commit:
cleaning up

mikkeschiren commented:
Updated documentation, fixed puppeteer-core into the image.

Example used for testing puppeteer in the container:

```
const puppeteer = require('puppeteer-core');
    puppeteer
      .launch({
        executablePath: '/usr/bin/chromium-browser',
        args: ['--no-sandbox', '--disable-dev-shm-usage'],
      })
      .then(async (browser) => {
         const page = await browser.newPage();
         await page.goto('https://www.thefarside.com/');
         await page.screenshot({path: 'farside.png'});
         await browser.close();

      });

```

mikkeschiren commented:
Just discovered that I broke something i sidekick when I rebuilt the container... I will be back if not someone else do it for me :-)
